hazelcast/src/main/java/com/hazelcast/ringbuffer/impl/RingbufferService.java

```diff
@@ -346,4 +360,133 @@ public void ensureQuorumPresent(String distributedObjectName, QuorumType require
         quorumService.ensureQuorumPresent(getQuorumName(distributedObjectName), requiredQuorumPermissionType);
     }
 
+    @Override
+    public Runnable prepareMergeRunnable() {
+        Map<Integer, List<RingbufferContainer>> partitionContainerMap = createHashMap(containers.size());
+        for (Entry<Integer, Map<ObjectNamespace, RingbufferContainer>> entry : containers.entrySet()) {
+            int partitionId = entry.getKey();
+            if (!partitionService.isPartitionOwner(partitionId)) {
+                continue;
+            }
+
+            List<RingbufferContainer> containerList = partitionContainerMap.get(partitionId);
+            if (containerList == null) {
+                containerList = new LinkedList<RingbufferContainer>();
+                partitionContainerMap.put(partitionId, containerList);
+            }
+            for (RingbufferContainer container : entry.getValue().values()) {
+                String serviceName = container.getNamespace().getServiceName();
+                // we just merge ringbuffer containers which are not used by other services
+                if (SERVICE_NAME.equals(serviceName) && !(getMergePolicy(container) instanceof DiscardMergePolicy)) {
+                    containerList.add(container);
+                }
+                container.cleanup();
+            }
+        }
+
+        // clear all items either owned or backup
+        reset();
```

Comment:
I guess this is tricky since now we removed all of the ringbuffers, including the ones belonging to map and cache. We should probably remove just the ones belonging to the ringbuffer service.

Reply:
At the moment the IMap and ICache PRs don't have any `EventJournal` implementation. I wanted to get everything merged first, before fixing that. I still have no clear picture in my mind, how the `EventJournal` should be merged (will make some examples of before/after first, so we can decide what is expected).